packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -45,14 +45,77 @@ abstract interface class WidgetStatesConstraint {
   bool isSatisfiedBy(Set<WidgetState> states);
 }
 
-// A private class, used in [WidgetStateOperators].
-class _WidgetStateOperation implements WidgetStatesConstraint {
-  const _WidgetStateOperation(this._isSatisfiedBy);
+@immutable
+sealed class _WidgetStateCombo implements WidgetStatesConstraint {
+  const _WidgetStateCombo(this.first, this.second);
```

Comment:
Since both `first` and `second` have the same type, do we make these required named arguments, to avoid mixups when adding new subclasses?